### PR TITLE
Add support for whitelisting columns from queries

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
@quinthar 

This is part of the change required for: https://github.com/Expensify/Expensify/issues/34018

This adds early support for blacklisting columns in a query. The intention is to build a subclass of the `DB` plugin that uses this functionality to allow queries that can't access certain DB columns. This functionality may change somewhat before it's finalized, but the intention with this version is to reduce the simultaneous number of moving pieces that depend on one another (i.e., testing this functionality requires coordinating changes across three git repositories).

Merging this change makes it easier to work on the dependent change in `Server-Expensify`.

### CODE REVIEW

**Comment by tylerkaraszewski** [comment]
Adding @cead22 as a reviewer in case @quinthar is busy today. This is an either/or review assignment, if either of you reviews this and is happy with it, go ahead and merge.

**Comment by quinthar** [comment]
Ya my schedule is a mess right now.

**Comment by tylerkaraszewski** [comment]
@cead22 Gave this to you!

File: sqlitecluster/SQLite.cpp [hunk 1]
```diff
@@ -526,3 +530,80 @@ size_t SQLite::getLastWriteChangeCount() {
...
+
+        // The following are *always* allowed in blacklist mode.
+        case SQLITE_SELECT:
+        case SQLITE_ANALYZE:
+        case SQLITE_FUNCTION:
+            return SQLITE_OK;
+            break;
+
+        // The following is the only special case where the blacklist actually applies.
+        case SQLITE_READ:
```
Comment (cead22) [inline sqlitecluster/SQLite.cpp]: What's the difference between SQLITE_SELECT and SQLITE_READ? I couldn't find it here https://www.sqlite.org/c3ref/c_alter_table.html or in the code base

**Comment by tylerkaraszewski** [comment]
A whiltelist wouldn't be much harder to build than a blacklist, it's just going to have a ton more entries.

File: sqlitecluster/SQLite.cpp (hunk 1 above)
Comment (tylerkaraszewski) [inline sqlitecluster/SQLite.cpp]: The actual difference isn't super clear from the docs, but it seems to be that `SQLITE_SELECT ` checks authorization to do a `SELECT` operation at all, but `SQLITE_READ` authorizes the caller to look at particular tables and columns.

**Comment by tylerkaraszewski** [comment]
@cead22 - comments addressed.

**Comment by tylerkaraszewski** [comment]
@cead22 - converted to whitelist. I'll re-work my Server-Expensify change to reflect that, as well.

**Pull request merged**